apps/OpenSpace/ext/launcher/src/sgctedit/windowcontrol.cpp

```diff
@@ -0,0 +1,521 @@
+                                                const QString* winColors, QWidget *parent)
+    : QWidget(parent)
+    , _nMonitors(nMonitors)
+    , _monIndex(monitorIndex)
+    , _index(windowIndex)
+    , _monitorResolutions(monitorDims)
+    , _colorsForWindows(winColors)
+{
+    createWidgets(parent);
+}
+
+void WindowControl::createWidgets(QWidget* parent) {
+    _windowDims = defaultWindowSizes[_index];
+    _size_x = new QLineEdit(
+        QString::fromUtf8(std::to_string(int(_windowDims.width())).c_str()), parent);
+    _size_y = new QLineEdit(
+        QString::fromUtf8(std::to_string(int(_windowDims.height())).c_str()), parent);
+    _offset_x = new QLineEdit(
+        QString::fromUtf8(std::to_string(int(_windowDims.x())).c_str()), parent);
+    _offset_y = new QLineEdit(
+        QString::fromUtf8(std::to_string(int(_windowDims.y())).c_str()), parent);
+    {
+        QIntValidator* validatorSize_x = new QIntValidator(10, _maxWindowSizePixels);
+        QIntValidator* validatorSize_y = new QIntValidator(10, _maxWindowSizePixels);
+        QIntValidator* validatorOffset_x = new QIntValidator(
+            -_maxWindowSizePixels,
+            _maxWindowSizePixels
+        );
+        QIntValidator* validatorOffset_y = new QIntValidator(
+            -_maxWindowSizePixels,
+            _maxWindowSizePixels
+        );
+        _size_x->setValidator(validatorSize_x);
+        _size_y->setValidator(validatorSize_y);
+        _offset_x->setValidator(validatorOffset_x);
+        _offset_y->setValidator(validatorOffset_y);
+    }
+    if (_nMonitors > 1) {
+        _comboMonitorSelect = new QComboBox(this);
+        _comboMonitorSelect->addItems(_monitorNames);
+        _comboMonitorSelect->setCurrentIndex(_monIndex);
+    }
+    _fullscreenButton = new QPushButton(this);
+    _fullscreenButton->setText("Set to Fullscreen");
+    _checkBoxWindowDecor = new QCheckBox("Window Decoration", this);
+    _checkBoxWindowDecor->setCheckState(Qt::CheckState::Checked);
+    _checkBoxWebGui = new QCheckBox("WebGUI only this window", this);
+    _checkBoxSpoutOutput = new QCheckBox("Spout Output", this);
+    _comboProjection = new QComboBox(this);
+    _comboProjection->addItems(_projectionTypes);
+
+    _comboQuality = new QComboBox(this);
+    _comboQuality->addItems(_qualityTypes);
+
+    {
+        _lineFovH = new QLineEdit("80.0", parent);
+        _lineFovV = new QLineEdit("50.534", parent);
+        QDoubleValidator* validatorFovH = new QDoubleValidator(-180.0, 180.0, 10);
+        _lineFovH->setValidator(validatorFovH);
+        QDoubleValidator* validatorFovV = new QDoubleValidator(-90.0, 90.0, 10);
+        _lineFovV->setValidator(validatorFovV);
+        _lineHeightOffset = new QLineEdit("0.0", parent);
+        QDoubleValidator* validatorHtOff= new QDoubleValidator(-1000000.0, 1000000.0, 12);
+        _lineHeightOffset->setValidator(validatorHtOff);
+    }
+
+    connect(
```

Comment:
See argument above for new `connect` methods.  For overloaded functions consider:
```
    connect(
        _clustersWidget, QOverload<Cluster::ID>::of(&ClustersWidget::killProcesses),
        this, QOverload<Cluster::ID>::of(&MainWindow::killAllProcesses)
    );
```